crates/bevy_pbr/src/render/pbr.wgsl

```diff
@@ -367,9 +374,47 @@ fn directional_light(light: DirectionalLight, roughness: f32, NdotV: f32, normal
 fn fetch_point_shadow(light_id: u32, frag_position: vec4<f32>, surface_normal: vec3<f32>) -> f32 {
     let light = point_lights.data[light_id];
 
+    let surface_to_light = light.position_radius.xyz - frag_position.xyz;
+
+    if ((light.flags & POINT_LIGHT_FLAGS_IS_SPOTLIGHT_BIT) != 0u) {
+        // construct the light view matrix
+        // view matrix z_axis is the reverse of transform.forward()
+        let fwd = -light.light_custom_data.xyz;
+        let distance_to_light = dot(fwd, surface_to_light);
+        let offset_position = -surface_to_light + (light.shadow_depth_bias * normalize(surface_to_light)) + (surface_normal.xyz * light.shadow_normal_bias) * distance_to_light;
+
+        // the construction of the up and right vectors needs to precisely mirror the code 
+        // in render/light.rs:spotlight_view_matrix
+        var sign = -1.0;
+        if (fwd.z >= 0.0) {
+            sign = 1.0;
+        }
```

Comment:
i did this because i had some issues with treatment of ``-0.0``, but that was a few iterations ago, i fixed some other issues since so i'm going to check again. But i noticed that the WGSL spec says ``sign()`` is "correctly rounded" and "doesn't specify a rounding mode". i don't know what that implies for treatment of the sign of -0.0 but maybe being explicit is better?

Reply:
If this is needed to be correct in the `-0.0` case, then that's fine. I was mostly just wondering why you didn't use the `sign()` built-in function.

Reply:
this is needed. i also had to use xz for the 2 components of the direction to preserve the z sign correctly (last commit).